packages/cosmos_ui_components/lib/components/template/cosmos_wallet_list_item.dart

```diff
@@ -1,53 +1,67 @@
+import 'package:cosmos_ui_components/components/cosmos_wallet_edit_button.dart';
 import 'package:cosmos_ui_components/components/gradient_avatar.dart';
+import 'package:cosmos_ui_components/cosmos_text_theme.dart';
 import 'package:cosmos_ui_components/cosmos_theme.dart';
 import 'package:flutter/material.dart';
 
 class CosmosWalletListItem extends StatelessWidget {
   final String name;
   final String address;
   final bool isSelected;
+  final bool isEditing;
   final VoidCallback onClicked;
+  final Function(String) onEditIconPressed;
```

Comment:
to make it more generic we should just allow to pass in the `suffix`. if one needs custom logic, it can be implemented outside of the component itself. Also, the `Icons.check_circle_outline` should be replaced with an asset from Figma. and it should not be visible by default, but only for currently selected wallet

Reply:
Updated